packages/core/src/physics/joint/HingeJoint.ts

```diff
@@ -0,0 +1,160 @@
+      swingOffset.copyFrom(value);
+    }
+    (<IHingeJoint>this._nativeJoint).setSwingOffset(swingOffset);
+  }
+
+  /**
+   * The connected anchor position.
+   * @remark If connectedCollider is set, this anchor is relative offset.
+   * Or the anchor is world anchor position.
+   */
+  get connectedAnchor(): Vector3 {
+    return this._connectedCollider.localPosition;
+  }
+
+  set connectedAnchor(value: Vector3) {
+    (<IHingeJoint>this._nativeJoint).setConnectedAnchor(value);
+  }
+
+  /**
+   * The current angle in degrees of the joint relative to its rest position.
+   */
+  get angle(): number {
+    return (<IHingeJoint>this._nativeJoint).getAngle();
+  }
+
+  /**
+   * The angular velocity of the joint in degrees per second.
+   */
+  get velocity(): Readonly<Vector3> {
+    return (<IHingeJoint>this._nativeJoint).getVelocity();
+  }
+
+  /**
+   * Enables the joint's limits. Disabled by default.
+   */
+  get useLimits(): boolean {
+    return (this._hingeFlags & HingeJointFlag.LimitEnabled) == HingeJointFlag.LimitEnabled;
+  }
+
+  set useLimits(value: boolean) {
+    if (value !== this.useLimits) {
+      this._hingeFlags |= HingeJointFlag.LimitEnabled;
+    }
+    (<IHingeJoint>this._nativeJoint).setHingeJointFlag(HingeJointFlag.LimitEnabled, value);
+  }
+
+  /**
+   * Enables the joint's motor. Disabled by default.
+   */
+  get useMotor(): boolean {
+    return (this._hingeFlags & HingeJointFlag.DriveEnabled) == HingeJointFlag.DriveEnabled;
+  }
+
+  set useMotor(value: boolean) {
+    if (value !== this.useMotor) {
+      this._hingeFlags |= HingeJointFlag.DriveEnabled;
+    }
+    (<IHingeJoint>this._nativeJoint).setHingeJointFlag(HingeJointFlag.DriveEnabled, value);
+  }
+
+  /**
+   * Enables the joint's spring. Disabled by default.
+   */
+  get useSpring(): boolean {
+    return this._useSpring;
+  }
+
+  set useSpring(value: boolean) {
+    this._useSpring = value;
+  }
+
+  /**
+   * The motor will apply a force up to a maximum force to achieve the target velocity in degrees per second.
+   */
+  get motor(): JointMotor {
+    return this._jointMonitor;
+  }
+
+  set motor(value: JointMotor) {
+    this._jointMonitor = value;
+    (<IHingeJoint>this._nativeJoint).setDriveVelocity(value.targetVelocity);
+    (<IHingeJoint>this._nativeJoint).setDriveForceLimit(value.forceLimit);
+    (<IHingeJoint>this._nativeJoint).setDriveGearRatio(value.gearRation);
+    (<IHingeJoint>this._nativeJoint).setHingeJointFlag(HingeJointFlag.DriveFreeSpin, value.freeSpin);
+  }
+
+  /**
+   * Limit of angular rotation (in degrees) on the hinge joint.
+   */
+  get limits(): JointLimits {
+    return this._limits;
+  }
+
+  set limits(value: JointLimits) {
+    this._limits = value;
+    if (this.useSpring) {
+      (<IHingeJoint>this._nativeJoint).setSoftLimit(value.min, value.max, value.stiffness, value.damping);
+    } else {
+      (<IHingeJoint>this._nativeJoint).setHardLimit(value.min, value.max, value.contactDistance);
+    }
```

Comment:
Here has a BUG:
1. set `useSpring  == false`
2. and then set `limits`
3. set `useSpring  == true`

Reply:
done